### DISCUSSION TITLE
How to view data in a topic?

### DISCUSSION THREAD

**Discussion by TrashFish123** [emqx/emqx#15410] [Q&A]
The data transmitted from my microcontroller is in JSON format. I want to view this data in a chart format. Currently, I can only subscribe through a client and view each data point individually. Instead, I'd like to have a graphical page, for instance, to display 'Temperature: 30' in a graphical interface.

---
*This issue was automatically translated from Chinese.*

**Comment by zmstone** [comment]
Hi @TrashFish123 
MQTT message data visualization is not a feature EMQX can support (at least not at the currently, 5.10).

You can maybe configure a data integration to export the data to a time series database such as influxdb, timescale, GreptimeDB or [Datalayers](https://docs.emqx.com/en/emqx/latest/data-integration/data-bridge-datalayers.html)

You can read more about data integrations here: https://docs.emqx.com/en/emqx/latest/data-integration/data-bridges.html
